src/tests/DownloadTests.cs

```diff
@@ -134,7 +134,7 @@ private async Task TestDownload(string purl, string targetFilename)
             {
                 try
                 {
-                    downloadTool.Download(packageUrl, tempDirectoryName).Wait();
+                    downloadTool.EnsureDownloadExists(packageUrl, tempDirectoryName).Wait();
```

Comment:
You should test the `ensureDownloadExists(url,null)` case as well

Reply:
added